receiver/sqlserverreceiver/config_windows.go

```diff
@@ -0,0 +1,19 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//go:build windows
+
+package sqlserverreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/sqlserverreceiver"
+
+import "fmt"
+
+func (cfg *Config) validateInstanceAndComputerName() error {
+	if cfg.InstanceName != "" && cfg.ComputerName == "" {
+		return fmt.Errorf("'instance_name' may not be specified without 'computer_name'")
+	}
+	if cfg.InstanceName == "" && cfg.ComputerName != "" {
+		return fmt.Errorf("'computer_name' may not be specified without 'instance_name'")
+	}
```

Comment:
AFAIU, both options are expected to be `""` for non-Windows, right? Those values pass this validation. So why do we need to run the validation on Windows only?

Reply:
`InstanceName` can be used on non-Windows as a part of a `WHERE` clause of the query to only receive metrics for a specific instance.

For Windows: The requirement is to specify both or neither.
For non-Windows: It can be any combination, though `ComputerName` is never checked or used. So `InstanceName` is optional on non-Windows.

Reply:
Ok, thanks for clarifying